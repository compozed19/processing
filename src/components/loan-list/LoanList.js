import React, { Component } from 'react'
import { withRouter } from "react-router-dom";
import './loan-list.css'
import Table from '@material-ui/core/Table';
import TableBody from '@material-ui/core/TableBody';
import TableCell from '@material-ui/core/TableCell';
import TableHead from '@material-ui/core/TableHead';
import TableRow from '@material-ui/core/TableRow';
import Paper from '@material-ui/core/Paper';
import Link from '@material-ui/core/Link';
import LocationIcon from '../../assets/Group5496.svg';
import LoanListChild from '../loan-list/LoanListChild';

class LoanList extends Component {
    constructor(props) {
       super(props)
    }
    // componentDidMount() {

    //   const data = this.props.getLoanList();
    //   return data;
    // }
    
    render() {
        const { loans } = this.props;
        return <div id="loanlist"><LoanListChild { ...loans } /></div> 
    }
}

export default LoanList;
