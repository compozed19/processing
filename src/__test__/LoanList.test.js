
import React from 'react';
import renderer from 'react-test-renderer';
import Enzyme from 'enzyme';
import LoanList from '../components/loan-list/LoanList';
import { shallow, mount } from 'enzyme';
import { shallowToJson } from 'enzyme-to-json';
import sinon from 'sinon';

import  {configure} from 'enzyme';
import Adapter from 'enzyme-adapter-react-16';
import configureMockStore from 'redux-mock-store';
const mockStore = configureMockStore();

configure({adapter : new Adapter()});


describe("LoanList page ", () => {
  let wrapper, loans;

  beforeEach(() => {
   let props = {
      loans: {
        "totalCount": 2,
        "loanList": [
          {
            "loanNumber": 19047528,
            "borrowerName": "Toto, Jojn",
            "subjectPropertyAddress": [
              "1700 Knoll Trail Dr, Dallas TX 75248"
            ],
            "currentMilestone": "Sent to Processing",
            "propertyType": "Res Owner Occupied",
            "loanAmount": 50058.12,
            "newMoney": 1058.21,
            "underwritingConditions": "Mortgage Statement",
            "loanOfficer": "Bobby Finken",
            "leadSource": "Internet",
            "fileStartDate": "09/04/2019",
            "processingActionItems": "",
            "taxRequestDate": "09/04/2019",
            "fileClosingDate": "09/04/2019",
            "lienHolder": "",
            "numberOfLienHolders": 0,
            "phoneNumber": "78412699",
            "email": "Toto@cazcreek.com",
            "transferee": "toto",
            "interestRate": 7.5,
            "numberOfSubjectProperties": 1
          },
        ]
      }
    }
  })

  // it('shows LoanListChild', () => {
  //   const store = mockStore();
  //   const loanListComponent = shallow(<LoanList store={store} />);
  //   LoanListChild = loanListComponent.dive().find(LoanListChild);
  //   expect(loanListComponent).to.exist
  // })

  xit('ComponentDidMount', () => {
    wrapper = shallow(<LoanList {...props} />)     
    expect(loans.loanList.length).toEqual(1)
  })
  xit('renders without crashing', () => {
    let tree;
    tree = shallow(<LoanList />);
    let div = tree.find('#loan');
    let table = tree.find('Table');

    expect(tree).toBeDefined();
    expect(div).toBeDefined();
  });

  xit('should render correctly with no props', () => {
    const component = mount(<LoanList />);
    expect(component).toMatchInlineSnapshot();

    // let data = '<div'
    // +'id="loanlist"'
    // +'>'
    // +'<LoanListChild />'
    // +'</div>';
    // expect(component).toEqual(data);

    // const tree = renderer.create(<LoanList/>).toJSON();
    // expect(tree).toMatchSnapshot();
  });

  xit('should render banner text correctly with given props', () => {
    let loans = {
      "totalCount": 2,
      "loanList": [
        {
          "loanNumber": 19047528,
          "borrowerName": "Toto, Jojn",
          "subjectPropertyAddress": [
            "1700 Knoll Trail Dr, Dallas TX 75248"
          ],
          "currentMilestone": "Sent to Processing",
          "propertyType": "Res Owner Occupied",
          "loanAmount": 50058.12,
          "newMoney": 1058.21,
          "underwritingConditions": "Mortgage Statement",
          "loanOfficer": "Bobby Finken",
          "leadSource": "Internet",
          "fileStartDate": "09/04/2019",
          "processingActionItems": "",
          "taxRequestDate": "09/04/2019",
          "fileClosingDate": "09/04/2019",
          "lienHolder": "",
          "numberOfLienHolders": 0,
          "phoneNumber": "78412699",
          "email": "Toto@cazcreek.com",
          "transferee": "toto",
          "interestRate": 7.5,
          "numberOfSubjectProperties": 1
        },
      ]
    }
    const component = shallow(<LoanList list={loans} />);

    expect(component).toMatchSnapshot();
  });

  xit('should render child component ', () => {
    let loans = {
      "totalCount": 2,
      "loanList": [
        {
          "loanNumber": 19047528,
          "borrowerName": "Toto, Jojn",
          "subjectPropertyAddress": [
            "1700 Knoll Trail Dr, Dallas TX 75248"
          ],
          "currentMilestone": "Sent to Processing",
          "propertyType": "Res Owner Occupied",
          "loanAmount": 50058.12,
          "newMoney": 1058.21,
          "underwritingConditions": "Mortgage Statement",
          "loanOfficer": "Bobby Finken",
          "leadSource": "Internet",
          "fileStartDate": "09/04/2019",
          "processingActionItems": "",
          "taxRequestDate": "09/04/2019",
          "fileClosingDate": "09/04/2019",
          "lienHolder": "",
          "numberOfLienHolders": 0,
          "phoneNumber": "78412699",
          "email": "Toto@cazcreek.com",
          "transferee": "toto",
          "interestRate": 7.5,
          "numberOfSubjectProperties": 1
        },
      ]
    }
    const component = shallow(<LoanList list={loans} />);
    let div = component.find('#loanlis');
    expect(div).toBeDefined();
    // expect(component).toMatchSnapshot();
  });
})