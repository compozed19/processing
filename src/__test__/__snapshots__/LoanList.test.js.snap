// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`LoanList page  should render banner text correctly with given props 1`] = `ShallowWrapper {}`;

exports[`LoanList page  should render child component  1`] = `ShallowWrapper {}`;

exports[`LoanList page  should render correctly with no props 1`] = `ShallowWrapper {}`;

exports[`LoanList page  should render correctly with no props 2`] = `
<div
  id="loanlist"
>
  <div
    className="container-fluid"
  >
    <br />
    <br />
    <br />
    <div
      className="MuiPaper-root MuiPaper-elevation1 alineaTable MuiPaper-rounded"
    >
      <table
        className="MuiTable-root"
        id="table"
      >
        <thead
          className="MuiTableHead-root"
        >
          <tr
            className="MuiTableRow-root MuiTableRow-head"
          >
            <th
              aria-sort={null}
              className="MuiTableCell-root MuiTableCell-head MuiTableCell-alignLeft"
              scope="col"
            >
              Loan
            </th>
            <th
              aria-sort={null}
              className="MuiTableCell-root MuiTableCell-head MuiTableCell-alignLeft"
              scope="col"
            >
              Borrower Name
            </th>
            <th
              aria-sort={null}
              className="MuiTableCell-root MuiTableCell-head MuiTableCell-alignLeft"
              scope="col"
            >
              Status
            </th>
            <th
              aria-sort={null}
              className="MuiTableCell-root MuiTableCell-head MuiTableCell-alignLeft"
              scope="col"
            >
              Subject Property
            </th>
            <th
              aria-sort={null}
              className="MuiTableCell-root MuiTableCell-head MuiTableCell-alignLeft"
              scope="col"
            >
              Property Type
            </th>
            <th
              aria-sort={null}
              className="MuiTableCell-root MuiTableCell-head MuiTableCell-alignLeft"
              scope="col"
            >
              Loan Amount
            </th>
            <th
              aria-sort={null}
              className="MuiTableCell-root MuiTableCell-head MuiTableCell-alignLeft"
              scope="col"
            >
              New $
            </th>
            <th
              aria-sort={null}
              className="MuiTableCell-root MuiTableCell-head MuiTableCell-alignLeft"
              scope="col"
            >
              Underwriting Conditions
            </th>
            <th
              aria-sort={null}
              className="MuiTableCell-root MuiTableCell-head MuiTableCell-alignLeft"
              scope="col"
            >
              Loan Officer
            </th>
            <th
              aria-sort={null}
              className="MuiTableCell-root MuiTableCell-head MuiTableCell-alignLeft"
              scope="col"
            >
              Lead Source
            </th>
            <th
              aria-sort={null}
              className="MuiTableCell-root MuiTableCell-head MuiTableCell-alignLeft"
              scope="col"
            >
              Interest Rate
            </th>
            <th
              aria-sort={null}
              className="MuiTableCell-root MuiTableCell-head MuiTableCell-alignLeft"
              scope="col"
            >
              Current Milestone
            </th>
            <th
              aria-sort={null}
              className="MuiTableCell-root MuiTableCell-head MuiTableCell-alignLeft"
              scope="col"
            >
              Date File Start
            </th>
            <th
              aria-sort={null}
              className="MuiTableCell-root MuiTableCell-head MuiTableCell-alignLeft"
              scope="col"
            >
              Processing action items
            </th>
            <th
              aria-sort={null}
              className="MuiTableCell-root MuiTableCell-head MuiTableCell-alignLeft"
              scope="col"
            >
              Tax request date
            </th>
            <th
              aria-sort={null}
              className="MuiTableCell-root MuiTableCell-head MuiTableCell-alignLeft"
              scope="col"
            >
              Closing Date
            </th>
            <th
              aria-sort={null}
              className="MuiTableCell-root MuiTableCell-head MuiTableCell-alignLeft"
              scope="col"
            >
              Lien Holder
            </th>
            <th
              aria-sort={null}
              className="MuiTableCell-root MuiTableCell-head MuiTableCell-alignLeft"
              scope="col"
            >
              Number of Lien holders
            </th>
            <th
              aria-sort={null}
              className="MuiTableCell-root MuiTableCell-head MuiTableCell-alignLeft"
              scope="col"
            >
              Phone number
            </th>
            <th
              aria-sort={null}
              className="MuiTableCell-root MuiTableCell-head MuiTableCell-alignLeft"
              scope="col"
            >
              Email
            </th>
            <th
              aria-sort={null}
              className="MuiTableCell-root MuiTableCell-head MuiTableCell-alignLeft"
              scope="col"
            >
              Transferee
            </th>
            <th
              aria-sort={null}
              className="MuiTableCell-root MuiTableCell-head MuiTableCell-alignLeft"
              scope="col"
            >
              Number of subject properties
            </th>
          </tr>
        </thead>
        <tbody
          className="MuiTableBody-root"
          id="tableBody"
        />
      </table>
    </div>
  </div>
</div>
`;
